### PR TITLE
Port to Python 3

### LINKED ISSUE #9: gwebsockets.protocol.BadRequestException: No WebSocket UPGRADE hdr: None
Sugar `shell.log` contains a traceback when a web activity is asked to stop;

```log
Traceback (most recent call last):
  File "/usr/lib/python3.7/dist-packages/gwebsockets/server.py", line 126, in _read_data_cb
    self._do_handshake()
  File "/usr/lib/python3.7/dist-packages/gwebsockets/server.py", line 78, in _do_handshake
    response = protocol.make_handshake(self._request)
  File "/usr/lib/python3.7/dist-packages/gwebsockets/protocol.py", line 231, in make_handshake
    headers.get('upgrade')))
gwebsockets.protocol.BadRequestException: No WebSocket UPGRADE hdr: None
```

Version tested was https://github.com/sugarlabs/gwebsockets/commit/240b189916cecf624b1ce3c6c12340a8cc357f9c.

Wireshark shows this HTTP request is received by gwebsockets when the activity starts, and there is no response from gwebsockets;
```log
GET / HTTP/1.1
Host: 127.0.0.1:35485
User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0 Safari/605.1.15
Origin: activity://com.garycmartin.moon
Pragma: no-cache
Cache-Control: no-cache
Upgrade: websocket
Connection: Upgrade
Sec-WebSocket-Key: iBBCmT+TO/nOjkFHd6IaPg==
Sec-WebSocket-Version: 13
Accept-Encoding: gzip, deflate
Accept-Language: en-AU
```

Enabling debug logging in gwebsockets shows;

```log
1580104500.084468 DEBUG gwebsockets: Got data, length 417
```

and this indicates the data is not fragmented.

### PR DESCRIPTION
**Pull request by quozl**
Fixes https://github.com/sugarlabs/gwebsockets/issues/9

Regression introduced 71e1e27 ("2to3 changes"), which was not fully
tested.

### CODE REVIEW

**Review by rhl-bthr — APPROVED** [review]
Reviewed till cfda09b

File: gwebsockets/server.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 if sys.version_info[0] < 3:
     from StringIO import StringIO
 else:
-    from io import StringIO
+    from io import BytesIO
 from collections import deque
```
Comment (srevinsaju) [inline gwebsockets/server.py]: This should possibly create a conflict. StringIO is not used, BytesIO is not imported in Python2, we can remove python2 version right?

**Comment by quozl** [comment]
It was easy to remove Python 2 support, but it turns out to be still needed, for systems that can run Python 2 Sugar activities.  Python 2 is still very much available from Linux distributions, despite the noise.  So I'll be rewriting these commits.

Of interest is that byte parsing in protocol.py is very much like https://github.com/concordusapps/tulip/blob/master/tulip/http/websocket.py which is a copy of where it came from.

**Comment by quozl** [comment]
I've thought some more about this.  gwebsockets is imported by Sugar only, not the Toolkit.  No system is likely to run both Sugar for Python 2 and Sugar for Python 3.  So support for Python 2 isn't needed at the same time as support for Python 3.

That makes this pull request ready for review.

**Review by rhl-bthr — APPROVED** [review]
Reviewed till bde4b18

**Pull request merged**